analytics.py

```diff
@@ -0,0 +1,289 @@
+     p. 445-453.
+    """
+    markList = []
+    if not mark: #If mark is empty, then you're computing the distance of all the points
+        shDistL =[]       #list of shortest distances
+
+        #now the points are numbered... so if the points
+        #have the same counter number attached also, then they
+        #are self-neighbors, but if num1 != num2, then they are
+        # coincident points, with distance = 0
+        printXonce = False
+        for num1, point in enumerate(points):
+            shortestDistance = math.inf
+            for num2, dpoint in enumerate(points):
+                if num1 != num2:
+                    if printXonce == False:
+                        print(point.x)
+                    epoint1 = (point.x,point.y)
+                    epoint2 = (dpoint.x,dpoint.y)
+                    dist = utils.euclidean_distance(epoint1, epoint2)  #changed input parameters because cannot pass in Point
+                    if(shortestDistance > dist):
+                        shortestDistance = dist
+                printXonce = True
+            #now add the shortest distance of that point before it moves on to a new point
+            shDistL.append(shortestDistance)
+       # print(shDistL)
+        sums = sum(shDistL)
+        mean_d = sums/len(shDistL)
+    #compute the average nearest neighbor distance of only those that share the mark
+    else:
+        for p in points:
+            if p.mark in mark:  #passed in a list of possible marks
+                markList.append(p)
+        shDistL =[]       #list of shortest distances
+
+        #now the points are numbered... so if the points
+        #have the same counter number attached also, then they
+        #are self-neighbors, but if num1 != num2, then they are
+        # coincident points, with distance = 0
+        for num1, point in enumerate(markList):
+            shortestDistance = math.inf
+            for num2, dpoint in enumerate(markList):
+                if num1 != num2:
+                    dist = utils.euclidean_distance((point.x,point.y), (dpoint.x,dpoint.y))
+                    if(shortestDistance > dist):
+                        shortestDistance = dist
+            #now add the shortest distance of that point before it moves on to a new point
+            shDistL.append(shortestDistance)
+        #print(shDistL)
+        sums = sum(shDistL)
+        mean_d = sums/len(shDistL)
+        print(mean_d)
+    return mean_d
+
+
+def minimum_bounding_rectangle(points):
+    """
+    Given a set of points, compute the minimum bounding rectangle.
+
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+
+    Returns
+    -------
+     : list
+       Corners of the MBR in the form [xmin, ymin, xmax, ymax]
+    """
+    # a minimum bounding rectangle would be on the extremes of x/y
+
+    xmin = math.inf
+    ymin = math.inf
+    xmax = -9999999999
+    ymax = -9999999999
```

Comment:
These should all swap to `math.inf`.  The numbers you are using are only large if we also account for unit (which we are not).